components/PlaybackTask.brs

```diff
@@ -1,253 +1,348 @@
+'
+' Return:
+'   Preroll AdPod if it exists
+'-----------------------------------------
+function getPreroll() as Object
+    ? "TRUE[X] >>> PlaybackTask::getPreroll(): " 
+
+    ads = m.raf.getAds()
+    if ads = invalid then return invalid
+    
+    result = invalid
+    for each adPod in ads
+        if adPod.rendersequence = "preroll"
+            result = adPod
+            exit for
+        end if
+    end for
+
+    return result
 end function
 
+'-------------------------------------------
+' General ad handler.  Takes care of seeing for a given pod, to play a truex ad
+' or play regular ads
+'
+' Return:
+'   false if playback should not be resumed yet (truex case mainly)
+'   true if playback should be resumed (non-ad or certain raf cases)
+'-----------------------------------------
 function handleAds(ads) as Boolean
-  resumePlayback = true
-
-  if ads <> invalid AND ads.ads.count() > 0
-    m.currentAdPod = ads
-    firstAd = ads.ads[0] 'Assume truex can only be first ad
-
-    if firstAd.adParameters <> invalid
-      m.truexAd = firstAd
-      m.truexAd.adParameters = parseJSON(m.truexAd.adParameters)
-      m.truexAd.renderSequence = ads.renderSequence
-      ads.ads.delete(0) ' Removes it from certain use cases due to raf
-
-      playTrueXAd()
-      resumePlayback = false
-    else ' Non-TrueX ads
-      hidePlayback()
-      watchedAd = m.raf.showAds(ads, invalid, m.adFacade) 'Takes thread ownership until complete or exit
-
-      if watchedAd
-        resumePlayback = true 
-      else
-        resumePlayback = false
-        exitPlayback() 
-      end if
+    resumePlayback = true
+
+    if ads <> invalid AND ads.ads.count() > 0
+        m.currentAdPod = ads
+        firstAd = ads.ads[0] 'Assume truex can only be first ad in a pod
+
+        if isTruexAd(firstAd)
+            data = {
+                adParameters: parseJSON(firstAd.adParameters),
+                renderSequence: ads.renderSequence
+            }            
+
+            ' Need to delete the ad from the pod which is referenced by raf so it plays
+            ' ads from the correct index when resulting in non-truex flows (eg. opt out)
+            ' If it is not deleted, this pod will attempt to play the truex ad placeholder
+            ' when it is passed into raf.showAds()
+            ads.ads.delete(0)
+
+            playTrueXAd(data)
+            resumePlayback = false
+        else ' Non-TrueX ads
+            hideContentStream()
+            watchedAd = m.raf.showAds(ads, invalid, m.adFacade) ' Takes thread ownership until complete or exit
+
+            if watchedAd
+                resumePlayback = true 
+            else
+                resumePlayback = false
+                exitContentStream() 
+            end if
+        end if
     end if
-  end if
 
-  return resumePlayback
+    return resumePlayback
+end function
+
+'-------------------------------------------
+' Helper to see if an ad is TrueX
```

Comment:
Add more color here that explains that our standard ad tag includes `<AdParameters>` and points to a get.truex.com DNS.